### ISSUE TITLE
Accept multiple target flags and build the targets concurrently

### ISSUE THREAD

**Issue by zmanji** [PyO3/maturin#1340]
Currently in order to cross compile wheels for many different targets I run something like:
```
maturin --release --zig --compatibility manylinux_2_28 --target x86_64-unknown-linux-gnu
maturin --release --zig --compatibility manylinux_2_28 --target aarch64-unknown-linux-gnu
```

I run these commands sequentially, and this is because I cannot specify multiple `--target` in a single command. If I specify multiple `--target` I get the following error:
```
error: The argument '--target <TRIPLE>' was provided more than once, but cannot be used multiple times

Usage: maturin build [OPTIONS] [ARGS]...

For more information try '--help'
```

However cargo normally accepts multiple `--target` flags. For example in a pure rust project I can do:
```
cargo build --release --target x86_64-unknown-linux-gnu --target aarch64-unknown-linux-gnu
```

and cargo will build both targets in parallel.

Would it be possible to enhance `maturin` to accept multiple target flags and pass them onto `cargo build`? Then instead of building the different wheels sequentially, you can instead build them in parallel which would hopefully speed up the overall process.

**Comment by messense** [comment]
Duplicate of #1190
